### PR TITLE
Slot New Hearing API

### PR DESCRIPTION
**Pull request by lomaxap**
Resolves merge conflicts with `slot_new_hearing` methods

### CODE REVIEW

File: spec/feature/hearing_schedule/daily_docket_spec.rb [hunk 1]
```diff
@@ -73,20 +73,5 @@
       # created a ticket that we can look into after February.
       # expect(page).to have_content("8:30 am")
     end
-
-    scenario "User can postpone a hearing" do
```
Comment (sharonwarner) [inline spec/feature/hearing_schedule/daily_docket_spec.rb]: Bad merge?

File: app/models/hearing.rb [hunk 2]
```diff
@@ -98,10 +98,11 @@ def current_issue_count
   end
   #:nocov:
 
-  def slot_new_hearing(hearing_day_id, _scheduled_time_unused, _appeal_unused)
+  def slot_new_hearing(hearing_day_id, hearing_location_attrs: nil, **_args)
     # These fields are needed for the legacy hearing's version of this method
     Hearing.create!(hearing_day_id: hearing_day_id,
                     scheduled_time: scheduled_time,
```
Comment (lomaxap) [inline app/models/hearing.rb]: we may actually need to pass scheduled time. On the daily docket, if a user changes the time, they may expect the postponed hearing to have that time.

File: spec/feature/hearing_schedule/daily_docket_spec.rb (hunk 1 above)
Comment (lomaxap) [reply spec/feature/hearing_schedule/daily_docket_spec.rb]: yea, I moved this method up and changed it a little and it didn't see that it was the same.

File: app/controllers/hearings_controller.rb [hunk 3]
```diff
@@ -111,6 +112,17 @@ def update_params_legacy
                                      ])
   end
 
+  def master_record_params
+    params.require("master_record_updated").permit(:id,
+                                                   time: [:h, :m, :offset],
```
[the remark this answers is not in the document]
Comment (sharonwarner) [reply app/controllers/hearings_controller.rb]: Got it. I'd like to carve out some time to refactor this after April 1st, so this is fine for now.

**Pull request merged**